google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spanner.spi.v1;
+
+import io.grpc.*;
+import io.grpc.ForwardingClientCall.SimpleForwardingClientCall;
+import io.grpc.ForwardingClientCallListener.SimpleForwardingClientCallListener;
+import io.opencensus.stats.*;
+import io.opencensus.tags.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import static com.google.cloud.spanner.spi.v1.SpannerRpcViews.*;
+
+
+public class HeaderInterceptor implements ClientInterceptor{
+
+    private static final Metadata.Key<String> SERVER_TIMING_HEADER_KEY =
+            Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER);
+    private static final Pattern SERVER_TIMING_HEADER_PATTERN = Pattern.compile(".*dur=(?<dur>\\d+)");
+    private static final Metadata.Key<String> GOOGLE_CLOUD_RESOURCE_PREFIX_KEY =
+            Metadata.Key.of("google-cloud-resource-prefix", Metadata.ASCII_STRING_MARSHALLER);
+    private static final Pattern GOOGLE_CLOUD_RESOURCE_PREFIX_PATTERN = Pattern.compile(".*projects/(?<project>\\w\\p{ASCII}+)/instances/(?<instance>\\w\\p{ASCII}+)/databases/(?<database>\\w\\p{ASCII}+)");
+
+    // Get the global singleton Tracer object.
+    private static final Tagger tagger = Tags.getTagger();
+    HeaderInterceptor(){}
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method,
+                                                               CallOptions callOptions, Channel next) {
+        return new SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+                TagContext tctx = getTagContext(headers, method.getFullMethodName());
+                super.start(new SimpleForwardingClientCallListener<RespT>(responseListener) {
+                    @Override
+                    public void onHeaders(Metadata metadata) {
+                        processHeader(metadata, tctx);
+                        super.onHeaders(metadata);
+                    }
+                }, headers);
+            }
+        };
+    }
+
+    private static void processHeader(Metadata metadata, TagContext tctx) {
```

Comment:
Can we make this function not static?
Please don't abbreviate variable names, `tctx -> tagContext`